pkg/gameserversets/controller.go

```diff
@@ -618,6 +619,14 @@ func computeStatus(list []*agonesv1.GameServer) agonesv1.GameServerSetStatus {
 		case agonesv1.GameServerStateReserved:
 			status.ReservedReplicas++
 		}
+
+		// TODO: Aggregates all Counters and Lists for all GameServer States that are not Deleting. This
+		// means that unlike player tracking below, this will aggregate values during GameServerStateCreating.
+		// Should I mimic player tracking below and only aggregate for cases Ready, Reserved, and Allocated?
```

Comment:
I think it depends on the use case. If I understand correctly, the major use case of the aggregated counters/lists is for Fleet autoscaling. So the values of the game servers that are not in ready reserved or allocated status ***should not*** be included. For example, if the autoscaler want to ensure there are at least 10 capacity for key A in the fleet, there is only one unhealthy game server in the fleet and it has 10 capacity for key A. Autoscaler should still scale up accordingly in this case.

Reply:
That makes sense, I'll update to only include status ready reserved or allocated.